mlflow/models.go

```diff
@@ -0,0 +1,55 @@
+package mlflow
+
+import (
+	"context"
+	"strconv"
+)
+
+type ModelService service
+
+type ModelVersionStatus string
+
+const (
+	ModelVersionStatusPending ModelVersionStatus = "PENDING_REGISTRATION"
+	ModelVersionStatusFailed  ModelVersionStatus = "FAILED_REGISTRATION"
+	ModelVersionStatusReady   ModelVersionStatus = "READY"
+)
+
+type ModelVersion struct {
+	Name                 string             `json:"name,omitempty"`
+	Version              string             `json:"version,omitempty"`
+	CreationTimestamp    int64              `json:"creation_timestamp,omitempty"`
+	LastUpdatedTimestamp int64              `json:"last_updated_timestamp,omitempty"`
+	UserID               string             `json:"user_id,omitempty"`
+	CurrentStage         string             `json:"current_stage,omitempty"`
+	Description          string             `json:"description,omitempty"`
+	Source               string             `json:"source,omitempty"`
+	RunID                string             `json:"run_id,omitempty"`
+	Status               ModelVersionStatus `json:"status,omitempty"`
+	StatusMessage        string             `json:"status_message,omitempty"`
+	Tags                 []*ModelVersionTag `json:"tags,omitempty"`
+	RunLink              string             `json:"run_link,omitempty"`
+	Aliases              []string           `json:"aliases,omitempty"`
+}
+
+type ModelVersionTag struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+func (s *ModelService) SetTag(ctx context.Context, name string, version int, key, value string) error {
```

Comment:
I think `version` should be a string since it's received as a string when fetching it.

Reply:
Had an even stronger and harder debate on this, went with the hand that said "why take a version **number** as a `string` if I can take it as an `int` ", but then when I used the API and had to convert, the other hand felt stronger :)
So again changing as suggested, thanks.